dgraph/cmd/alpha/run.go

```diff
@@ -746,7 +749,13 @@ func run() {
 			}
 			numShutDownSig++
 			glog.Infoln("Caught Ctrl-C. Terminating now (this may take a few seconds)...")
-			if numShutDownSig == 3 {
+
+			switch {
+			case atomic.LoadUint32(&initDone) < 2:
```

Comment:
condition is < 2 because you have two checkpoints that need to be reached to consider it initialized?

Reply:
Yes. One checkpoint in serve(..) function and the other one is after raft nodes are initialized.